sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/ApproximateUniqueTest.java

```diff
@@ -102,6 +113,20 @@ private static void verifyEstimate(final long uniqueCount,
                    + error + "%, MaxError=" + maxError + "%.", error < maxError);
   }
 
+  private static Matcher<Long> estimateIsWithinRangeFor(final long uniqueCount,
+                                                        final int sampleSize) {
+    if (uniqueCount <= sampleSize) {
+      return is(uniqueCount);
+    } else {
+      long maxError = (long) Math.ceil(2.0 * uniqueCount / Math.sqrt(sampleSize));
```

Comment:
Just checking: is this a probabilistic guarantee (holds 99% of the time), or a hard guarantee (holds 100% of the time)? Wouldn't want the test to be flaky.

Reply:
Good point. I didn't check. In the case of these tests, the input does not change run to run. Once it passes it should always pass. At the least it won't be flacky.

Reply:
I could not find where the error bound comes from or what the cutoff probability is. The estimate is roughly `k * hash_key_space / (max_hash - k_th_largest_hash)`. ApproximateUnique is likely to be replaced by ApproximateDistinct which estimates using HyperLogLog. The test are not flacky as the input is deterministic.